corehq/apps/userreports/models.py

```diff
@@ -350,6 +350,27 @@ def get_es_index_settings(self):
         es_index_settings.pop('doc_type')
         return {"settings": es_index_settings}
 
+    def get_case_type_or_xmlns_filter(self):
+        def _get_property_value(config_filter, prop_name):
+            if (config_filter.get('type') != 'boolean_expression'
+                    or config_filter['operator'] != 'eq'):
+                return None
+            expression = config_filter['expression']
+            if expression['type'] == 'property_name' and expression['property_name'] == prop_name:
```

Comment:
Does this rely on the fact that there will only be one boolean filter that checks `property_name` to a value? What if there is an additional filter that's not case_type/xmlns filter?

Reply:
If it's a more complex filter, it should return `None` and it will not try to filter anything so it would go through all forms or cases (as it does today).